src/Sfa.Tl.Matching.Application/Services/ReferralService.cs

```diff
@@ -39,38 +41,90 @@ public class ReferralService : IReferralService
+
+            //you could use DidNotReceived for hen_NotificationsApi_SendEmail_Is_Not_Called()
+
+            //Email subject and replyto - if not in use please remove
+            //DefaultReplyToAddress and ReplyToAddress - remove - if this is not working just remove it for now
+
+            // Do we need to log here that we didn't find any suitable template?
+            //if (emailTemplate == null)
+
+
+            // await _referralService.SendProviderReferralEmail(id);
+            //This need to happen before we redirect to Email Sent Screen i.e. in the post of CheckAnswers
+            //Also you might want to move this call to Opportunity service
+
+            //Use GetNumberOfPlacements everywhere
+
+            var emailTemplate = await GetEmailTemplate(EmployerReferralEmailTemplateName);
+
+            var referrals = await GetEmployerReferrals(opportunityId);
+
+            var employer = referrals.FirstOrDefault();
+
+            if (employer == null)
+            {
+                return;
+            }
+
+            var toAddress = employer.EmployerContactEmail;
+
+            var numberOfPlacements = GetNumberOfPlacements(employer.PlacementsKnown, employer.Placements);
+
+            var tokens = new Dictionary<string, string>
             {
-                new EmailPlaceholderDto {Key = "providers_list", Value = sb.ToString()}
+                { "employer_contact_name", employer.EmployerContact },
+                { "employer_business_name", employer.EmployerName },
+                { "employer_contact_number", employer.EmployerContactPhone },
+                { "employer_contact_email", employer.EmployerContactEmail },
+                { "employer_postcode", employer.Postcode },
+                { "number_of_placements", numberOfPlacements },
+                { "route", employer.RouteName },
+                { "job_role", employer.JobTitle }
             };
-            */
+
+            var sb = new StringBuilder();
+
+            foreach (var referral in referrals)
+            {
+                sb.AppendLine($"# {referral.ProviderName}");
+                sb.AppendLine($"{referral.ProviderVenuePostcode}");
+                sb.AppendLine($"Contact name: {referral.ProviderPrimaryContact}");
+                //sb.AppendLine($"Telephone: {referral.ProviderPrimaryContactPhone}");
+                sb.AppendLine($"Email: {referral.ProviderPrimaryContactEmail}");
+                sb.AppendLine("Has students learning: ");
+
+                foreach (var qualificationShortTitle in referral.QualificationShortTitles)
+                {
+                    sb.AppendLine($"* {qualificationShortTitle}");
+                }
+                sb.AppendLine(""); //Need a blank line, otherwise the next heading won't be formatted
+            }
+
+            tokens.Add("providers_list", sb.ToString());
+
+            await _emailService.SendEmail(emailTemplate.TemplateName,
```

Comment:
Might be worth passing in DTO

Reply:
You're the second person to say that, but I see it as a method with parameters, not something that is passing data around. Can consider as a future refactoring.